src/AWS.Deploy.Orchestration/CdkProjectHandler.cs

```diff
@@ -168,27 +196,55 @@ public async Task<bool> DetermineIfCDKBootstrapShouldRun()
             return false;
         }
 
-        private async Task CheckCdkDeploymentFailure(CloudApplication cloudApplication, DateTime deploymentStartDate)
+        private async Task<string> RetreiveStackId(CloudApplication cloudApplication, CancellationToken cancellationToken)
+        {
+            // Do an initial delay to avoid a race condition of the status being checked before the deployment has kicked off.
+            await Task.Delay(TimeSpan.FromSeconds(3));
```

Comment:
This isn't safe logic. If the CDK project involves uploading lots of data to S3 for example like the deployment bundle for Beanstalk or S3 files for Blazor this could take an undetermined amount of time.

Reply:
Removed